src/client.rs

```diff
@@ -17,15 +15,14 @@ use anyhow::{anyhow, bail};
 /// If successful, returns a tuple of a message stream and a sender,
 /// which may be used to receive and send messages respectively.
 pub async fn connect(
-    address: impl Into<String>,
+    server: impl tokio::net::ToSocketAddrs,
+    host: impl Into<String>,
```

Comment:
Maybe call this virtualhost, just to make it more clear what it represent and differentiate from the name "server"

Reply:
yes, It would be more appropriate to rename it as "virtualhost".

Reply:
I have been digging around the internet a bit, and it seems that most clients I find, doesn't support setting the virtualhost. Is that not a commonly supported feature on the server side? And, if that is not a widely used feature, doesn't it make sense to add a simplified connect method, where it defaults to the address? 

I'm just worried that this gets confusing for the regular use case. But I don't think we should sacrifice correctness either.